### PR TITLE
Allow searching for reports with ref: prefix in postcode field

### PR DESCRIPTION
**Pull request by davea**
Searches the `id` and `external_id` fields.

This takes you directly to the report if only a single report matches, otherwise
a disambiguation page is shown.

For mysociety/fixmystreetforcouncils#16

### CODE REVIEW

File: perllib/FixMyStreet/App/Controller/Around.pm [hunk 1]
```diff
@@ -38,6 +38,11 @@ sub index : Path : Args(0) {
     # Check if we have a partial report
     my $partial_report = $c->forward('load_partial');
 
+    # Check if the user is searching for a report by ID
+    if ( $c->get_param('pc') =~ /^\s*ref:(\d+)\s*$/ ) {
```
Comment (dracos) [inline perllib/FixMyStreet/App/Controller/Around.pm]: Could this also, for a UK cobrand, allow searching without `ref:` (as we don't have numeric zipcodes)?

File: perllib/FixMyStreet/App/Controller/Report.pm [hunk 2]
```diff
@@ -289,6 +289,24 @@ sub map : Path('') : Args(2) {
     $c->res->body($image->{data});
 }
 
+sub lookup_by_ref : Private {
```
Comment (dracos) [inline perllib/FixMyStreet/App/Controller/Report.pm]: As this is part of a front page search (and only sets stash stuff on success), perhaps better in Around.pm? Shrug.

**Comment by dracos** [comment]
I wondered if this should share more with the multiple location search results thing, given it's sort of the same type of thing, but seems fine like this :)

File: perllib/FixMyStreet/App/Controller/Around.pm (hunk 1 above)
Comment (davea) [inline perllib/FixMyStreet/App/Controller/Around.pm]: Good idea, done in 9654c64296bd31e8d37ef8f6f59c897245ff1f20.

File: perllib/FixMyStreet/App/Controller/Report.pm (hunk 2 above)
Comment (davea) [inline perllib/FixMyStreet/App/Controller/Report.pm]: Yep, have moved it.

**Comment by davea** [comment]
I thought about that, but decided keeping it fairly small and separate was the way to go.

**Pull request merged**